app/presenters/ht_download_presenter.rb

```diff
@@ -93,4 +97,15 @@ def show_institution_name
 
     link_to institution_name, ht_institution_path(inst_code)
   end
+
+  def show_pages
```

Comment:
See comments elsewhere about pages & Ransack; not sure what the best option here is.

Reply:
To the `ht_download` model you can start with a custom ransacker to translate the special case value of NULL in the table.

```
ransacker :pages
  Arel.sql("COALESCE(#{table_name}.pages, 'all')")
end
```

It doesn't take into account the possible "unknown" case represented in the presenter, where it's a partial download. That would be a more complex Arel operation (using Arel's equivalent of SQL `IF` perhaps, in place of the `'all'` literal?).